src/minimum_runtime_check.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @license
+ * Copyright 2024 The Emscripten Authors
+ * SPDX-License-Identifier: MIT
+ */
+
+#if ASSERTIONS
+
+// "30.0.0" -> 300000
+function humanReadableVersionToPacked(str) {
+  str = str.split('-')[0]; // Remove any trailing part from e.g. "12.53.3-alpha"
+  var vers = str.split('.').slice(0, 3);
+  while(vers.length < 3) vers.push('00');
+  vers = vers.map((n, i, arr) => n.padStart(2, '0'));
+  return vers.join('');
+}
+// 300000 -> "30.0.0"
+var packedVersionToHumanReadable = n => [n / 10000 | 0, (n / 100 | 0) % 100, n % 100].join('.');
+
+var TARGET_NOT_SUPPORTED = 2147483647;
+
+var currentNodeVersion = typeof process !== 'undefined' && process?.versions?.node ? humanReadableVersionToPacked(process.versions.node) : TARGET_NOT_SUPPORTED;
+if (currentNodeVersion < TARGET_NOT_SUPPORTED && {{{ MIN_NODE_VERSION }}} == TARGET_NOT_SUPPORTED) throw new Error('not compiled for this environment (did you build to HTML and try to run it not on the web, or set ENVIRONMENT to something - like node - and run it someplace else - like on the web?)');
+if (currentNodeVersion < {{{ MIN_NODE_VERSION }}}) throw new Error(`This emscripten-generated code requires node v${ packedVersionToHumanReadable({{{ MIN_NODE_VERSION }}}) } (detected v${packedVersionToHumanReadable(currentNodeVersion)})`);
+
+var currentSafariVersion = typeof navigator !== 'undefined' && navigator?.userAgent?.includes("Safari/") && navigator.userAgent.match(/Version\/(\d+\.?\d*\.?\d*)/) ? humanReadableVersionToPacked(navigator.userAgent.match(/Version\/(\d+\.?\d*\.?\d*)/)[1]) : TARGET_NOT_SUPPORTED;
+if (currentSafariVersion < {{{ MIN_SAFARI_VERSION }}}) throw new Error(`This emscripten-generated code requires Safari v${ packedVersionToHumanReadable({{{ MIN_SAFARI_VERSION }}}) } (detected v${currentSafariVersion})`);
+
+var currentFirefoxVersion = typeof navigator !== 'undefined' && navigator?.userAgent?.match(/Firefox\/(\d+(?:\.\d+)?)/) ? parseFloat(navigator.userAgent.match(/Firefox\/(\d+(?:\.\d+)?)/)[1]) : TARGET_NOT_SUPPORTED;
```

Comment:
These days you can do `globalThis.navigator?.userAgent?.match` avoiding the need for the `typeof` check.  It makes it a little smaller and easier to read.

See #25375 and #25381.

Reply:
Oh I see, we know the globalThis exists on supported browser, but here we are trying to explicitly detect unsupported ones.  That makes sense. 

I guess you should do check for `typeof globalThis` first before the other checks?  lgtm either way though